src/core/dot.rs

```diff
@@ -46,71 +42,54 @@ where
-                                    random_color,
-                                    weight.safe_debug()
-                                )
-                                .as_str(),
-                            ),
-                        ]
-                        .join("\n")
-                    }),
-                ]
-                .join("\n")
-            });
-
-    let vertices =
-        hypergraph
-            .vertices
-            .iter()
-            .enumerate()
-            .fold(String::new(), |acc, (index, (weight, _))| {
-                [
-                    acc,
-                    indent(
-                        format!(
-                            r#"{} [label="{:?}"{}];"#,
-                            index,
-                            weight.safe_debug(),
-                            // Inject peripheries for unaries.
-                            match partitioned_hyperedges.1.get(&index) {
-                                Some(weight) => format!(", peripheries={}", weight.len() + 1),
-                                None => String::new(),
-                            }
-                        )
-                        .as_str(),
-                    ),
-                ]
-                .join("\n")
-            });
-
-    // Return the rendered graph as String.
-    [
-        String::from("digraph {"),
-        indent("edge [penwidth=0.5, arrowhead=normal, arrowsize=0.5, fontsize=8.0];"),
-        indent("node [color=gray20, fontsize=8.0, fontcolor=white, style=filled, shape=circle];"),
-        indent("rankdir=LR;"),
-        vertices,
-        non_unary_hyperedges,
-        String::from("}"),
-    ]
-    .join("\n")
+    let mut output = Vec::new();
+    {
+        let mut writer = dot_writer::DotWriter::from(&mut output);
+        let mut graph = writer.digraph();
+
+        graph
+            .edge_attributes()
+            .set_pen_width(0.5)
+            .set_arrow_head(dot_writer::ArrowType::Normal)
+            .set_arrow_size(0.5)
+            .set_font_size(8.0);
+
+        graph
+            .node_attributes()
+            .set_color(dot_writer::Color::Gray20)
+            .set_font_size(8.0)
+            .set_font_color(dot_writer::Color::White)
+            .set_style(dot_writer::Style::Filled)
+            .set_shape(dot_writer::Shape::Circle);
+
+        graph.set_rank_direction(dot_writer::RankDirection::LeftRight);
+
+        for (index, (weight, _)) in hypergraph.vertices.iter().enumerate() {
+            let mut node = graph.node_named(&index.to_string());
+            node.set_label(&format!("{:?}", weight.safe_debug()));
+            if let Some(weight) = partitioned_hyperedges.1.get(&index) {
+                node.set("peripheries", &(weight.len() + 1).to_string(), false);
+            }
+        }
+
+        for (vertices, weights) in partitioned_hyperedges.0.iter() {
+            if vertices.len() < 2 {
+                continue; // can't do an edge if zero or one vertices to join!
```

Comment:
The zero issue - i.e. empty set - is discussed in #15. However, unaries - hyperedges with only on vertex - should be rendered properly. Is it the case?

Reply:
Yes ones with just one vertex are still rendered. (FYI I've just pushed a small refactor to the partitioning code which I think makes that a bit easier to read, but shouldn't change the functionality).